modules/programs/senpai.nix

```diff
@@ -61,13 +97,34 @@ in {
         <manvolnum>5</manvolnum></citerefentry>.
       '';
     };
+    extraConfig = mkOption {
+      type = types.attrs;
+      default = { };
+      description = ''
+        Options that should be appended to the senpai configuration file.
+      '';
+    };
   };
 
   config = mkIf cfg.enable {
+    assertions = with cfg.config; [
+      {
+        assertion = !isNull password-cmd -> isNull password;
+        message = "senpai: password-command overrides password!";
+      }
+      {
+        # TODO: remove me for the next release
+        assertion = isNull no-tls;
```

Comment:
If setting the option at all causes an error, we might as well remove it. Look into `mkRemovedOptionModule` or `mkChangedOptionModule`